test/integration/apiserver/discovery/discovery_test.go

```diff
@@ -285,188 +222,437 @@ func TestAggregatedAPIServiceDiscovery(t *testing.T) {
 	require.NoError(t, WaitForGroups(ctx, client, basicTestGroup))
 }
 
-// Shows that the following sequence is handled correctly:
-// 1. Create an APIService
-// - Check that API service is in discovery doc
-// 2. Create CRD with the same GroupVersion as APIService
-// 3. Delete APIService
-// - Check that API service is removed from discovery
-// 4. Update CRD
-// -  Check that CRD is in discovery document
-func TestOverlappingCRDAndAPIService(t *testing.T) {
-	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.AggregatedDiscoveryEndpoint, true)()
+func runTestCases(t *testing.T, cases []testCase) {
```

Comment:
Should this be a method of `testCase` that returns an error?

Reply:
i.e.
```suggestion
func (tc testCase) Run() error {
```

Reply:
That's nit by the way, the framework is great and totally reasonable for the current release stage.